eslint.config.js

```diff
@@ -0,0 +1,135 @@
+import reactHooksPlugin from 'eslint-plugin-react-hooks';
+import reactPlugin from 'eslint-plugin-react';
+import tsPlugin from '@typescript-eslint/eslint-plugin';
+import tsParser from '@typescript-eslint/parser';
+import unusedImports from 'eslint-plugin-unused-imports';
+
+export default [
+  {
+    // Global settings
+    languageOptions: {
+      ecmaVersion: 'latest',
+      sourceType: 'module',
+      parser: tsParser,
+      parserOptions: {
+        ecmaFeatures: {
+          jsx: true,
+          legacyDecorators: true,
+        },
+      },
+      globals: {
+        // Browser globals
+        window: 'readonly',
+        document: 'readonly',
+        navigator: 'readonly',
+        location: 'readonly',
```

Comment:
setting some of these globals to readonly feels like it could be problematic. in practice, modern browsers allow assignment to window.location which is functionally the same as window.location.assign()

Reply:
I would argue that it's a bad practice to use that over location.assign and the linter should warn you about that

Reply:
I think the best practice is actually to use this lib https://github.com/sindresorhus/globals/blob/main/globals.json I think I was just trying to avoid another dependency but we should just use that